syntax = "proto2";

package SimpleChat;

message ChatTarget {
    required string user_name = 1;
    optional int32 user_id = 2;
}

message ChatMessage {
    required ChatTarget from = 1;
    required string text = 2;
    required int64 timestamp = 3;

    optional ChatTarget target = 4;
}

message ChatAuthorize {
    required ChatTarget from = 1;
    required string password = 2;
}

enum CommandType {
    MOTD = 0;
    KICK = 1;
    MUTE = 2;
    UNKNOWN = 3;
}

message ChatCommand {
    required ChatTarget from = 1;
    required CommandType type = 2;
    repeated string arguments = 3;
}